pycross/private/tools/poetry_translator.py

```diff
@@ -32,26 +31,36 @@ class MismatchedVersionException(Exception):
     pass
 
 
+# Not using cached_property to support python 3.7
+CONSTRAINT_CACHE_V = "__constraint_cache"
+MARKER_WITH_EXTRA_CACHE_V = "__marker_without_extra"
 @dataclass
 class PoetryDependency:
     name: str
     spec: str
     marker: Optional[str]
 
-    @cached_property
     def constraint(self):
-        return parse_constraint(self.spec)
+        CONSTRAINT_CACHE_V = "__constraint_cache"
```

Comment:
Honestly I would just leave the property uncached and see how it goes. I doubt my decision to add `@cached_property` was based on any data.